snuba/consumers/consumer_builder.py

```diff
@@ -217,10 +259,14 @@ def __build_consumer(
                 commit_retry_policy=self.__commit_retry_policy,
             )
 
-        return StreamProcessor(consumer, self.raw_topic, strategy_factory, IMMEDIATE)
+        raw_topic_spec = KafkaTopicSpec(SnubaTopic(self.raw_topic.name))
+        physical_topic = ArroyoTopic(raw_topic_spec.get_physical_topic_name(slice_id))
+
+        return StreamProcessor(consumer, physical_topic, strategy_factory, IMMEDIATE)
```

Comment:
I think this is going to break all CLI topic overrides since you are attempting to remap it to something else in here.

Reply:
Please ensure that all the way of mapping of commit log topic, replacement topic and raw topic is being handled the same. We are solving the same thing for all 3.